### PR TITLE
Add pip cache in KPOps runner action

### PR DESCRIPTION
**Pull request by raminqaf**

### CODE REVIEW

File: actions/kpops-runner/action.yaml [hunk 1]
```diff
@@ -34,10 +34,20 @@ inputs:
   token:
     description: "secrets.GITHUB_TOKEN, needed for setup-helm action if helm-version is set to latest"
     required: false
+  python-version:
+    description: "Python version"
+    required: false
+    default: "3.10.6"
```
Comment (disrupted) [inline actions/kpops-runner/action.yaml]:
```suggestion
    default: "3.11"
```
3.11 for better tracebacks
patch level doesn't matter for the default imo

File: actions/kpops-runner/action.yaml [hunk 2]
```diff
@@ -34,10 +34,20 @@ inputs:
   token:
     description: "secrets.GITHUB_TOKEN, needed for setup-helm action if helm-version is set to latest"
     required: false
+  python-version:
```
Comment (disrupted) [inline actions/kpops-runner/action.yaml:37]: can you move this up above `kpops-version`?

File: actions/kpops-runner/action.yaml [hunk 3]
```diff
@@ -34,10 +34,20 @@ inputs:
...
+    description: "Python version"
```
Comment (disrupted) [inline actions/kpops-runner/action.yaml:38]: description should match the readme

File: actions/kpops-runner/README.md [hunk 4]
```diff
@@ -16,6 +16,7 @@ This action runs KPOps with the given config.
 | kpops-version     |    ❌    |    latest     | string | KPOps version to install                                                                                                                      |
 | helm-version      |    ❌    |    latest     | string | Helm version to install                                                                                                                       |
 | token             |    ❌    |    latest     | string | secrets.GITHUB_TOKEN, needed for setup-helm action if helm-version is set to latest                                                           |
+| python-version             |    ❌    |    "3.10.6"     | string | Python version to install
```
Comment (disrupted) [inline actions/kpops-runner/README.md]: please fix the table formatting (either manually or using dprint/prettier)

**Review by disrupted — CHANGES_REQUESTED** [review]

File: actions/kpops-runner/README.md [hunk 5]
```diff
@@ -16,6 +16,7 @@ This action runs KPOps with the given config.
 | kpops-version     |    ❌    |    latest     | string | KPOps version to install                                                                                                                      |
 | helm-version      |    ❌    |    latest     | string | Helm version to install                                                                                                                       |
 | token             |    ❌    |    latest     | string | secrets.GITHUB_TOKEN, needed for setup-helm action if helm-version is set to latest                                                           |
+| python-version             |    ❌    |    "3.x"     | string | Python version to install (Defaults to the latest stable version of Python for specified major version)
```
Comment (disrupted) [inline actions/kpops-runner/README.md:19]: with `3.x` this might break when Python 3.12 is released and kpops is not updated yet

**Review by disrupted — CHANGES_REQUESTED** [review]

File: actions/kpops-runner/README.md (hunk 5 above)
Comment (raminqaf) [reply actions/kpops-runner/README.md:19]: How about 3.11.x?

**Review by sujuka99 — APPROVED** [review]
LGTM

File: actions/kpops-runner/README.md (hunk 5 above)
Comment (disrupted) [reply actions/kpops-runner/README.md]: do you know what's the difference to 3.11?

**Review by disrupted — APPROVED** [review]

**Pull request merged**